model/channel.go

```diff
@@ -188,6 +201,18 @@ func (o *Channel) IsValid() *AppError {
 	return nil
 }
 
+func (o *GetChannelsOptions) IsValid() *AppError {
+	if reflect.TypeOf(o.UserIds).String() != "[]string" {
```

Comment:
`IsValid` is a good idea, Can `o.UserIds` or `o.ChannelTypes` be anything other than `[]string` ?
I'd add checking `page` and `perPage` here.